File: Keeper.client/src/models/Vault.js
import { RepoItem } from "./RepoItem.js";

export class Vault extends RepoItem {
  constructor(data) {
    super(data)
    this.name = data.name
    this.description = data.description
    this.img = data.img
    this.creator = data.creator
    this.isPrivate = data.isPrivate
  }
}

File: Keeper/Models/Vault.cs


namespace Keeper.Models;

public class Vault : RepoItem<int>
{
    public string CreatorId { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public string Img { get; set; }
    public Profile Creator { get; set; }
    public bool? isPrivate { get; set; }
    
}
